File: backend/app/models/item.py
from sqlalchemy import Column, Integer, String, Float, ForeignKey
from app.core.database import Base

class Item(Base):
    __tablename__ = "items"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, nullable=False)
    description = Column(String, nullable=True)
    price = Column(Float, nullable=False)
    owner_id = Column(Integer, ForeignKey("users.id"), nullable=False)

File: frontend/src/types/item.ts
 export interface Item {
  id: number
  title: string
  description?: string
  price: number
  owner_id: number
  created_at: string
  updated_at: string
}

 export interface ItemListProps {
  refreshTrigger: number
}

export interface CreateItemProps {
  onItemCreated: () => void
}
